django/db/models/deletion.py

```diff
@@ -469,11 +474,25 @@ def delete(self):
                     deleted_counter[qs.model._meta.label] += count
 
             # update fields
-            for model, instances_for_fieldvalues in self.field_updates.items():
-                for (field, value), instances in instances_for_fieldvalues.items():
+            for (field, value), instances_list in self.field_updates.items():
+                updates = []
+                objs = []
+                for instances in instances_list:
+                    if (
+                        isinstance(instances, models.QuerySet)
+                        and instances._result_cache is None
+                    ):
+                        updates.append(instances)
+                    else:
+                        objs.extend(instances)
+                if updates:
+                    combined_updates = reduce(or_, updates)
+                    combined_updates.update(**{field.name: value})
+                if objs:
```

Comment:
Do we need this branch? :thinking: All built-in handlers use non-evaluated querysets returned by `related_objects()`, so tests work without it:
```python
# update fields
for (field, value), instances_list in self.field_updates.items():
    updates = reduce(or_, instances_list)
    updates.update(**{field.name: value})
```

Reply:
Not on MySQL where the `can_defer_constraint_checks` feature flag is false. Try running the tests on MySQL and you'll see that you can end up here.

Reply:
You're right. Thanks for the clarification :+1: